src/scripts/sidebar.js

```diff
@@ -0,0 +1,25 @@
+window.addEventListener("load", () => {
+  const sidebarLinks = document.querySelectorAll(".link-sidebar");
+  sidebarLinks.forEach((link) => {
+    link.addEventListener("click", (clicked) => {
+      sidebarLinks.forEach((link) => link.classList.remove("active"));
+      findPairAndAddActive(clicked.target.attributes.href.nodeValue);
+    });
+  });
+
+  // if a user enters page with an anchor tag hash
+  function checkStoriesHash() {
+    if (window.location.hash.startsWith("#stories")) {
+      findPairAndAddActive(window.location.hash);
+    }
+  }
+
+  // find link pair and add active class to both links
+  function findPairAndAddActive(hash) {
+    linkPair = document.querySelectorAll("a[href='" + hash + "']");
+    linkPair.forEach((link) => link.classList.add("active"));
+  }
+
+  // call the function once on page load to check for the initial hash
+  checkStoriesHash();
+});
```

Comment:
@machikoyasuda I didn't see this file get any updates, still seeing the behavior I described above.

Reply:
Sorry! Now this is all in: https://github.com/cal-itp/calitp.org/pull/380/commits/c75ab7c04d9fab5594f3fbd69726e8b8319b8540 - tested it going forwards (clicking a Topic or Agency) and backwards (clicking Back)

Reply:
This is much better! There is still one small change I think we need:

When the hash is removed entirely from the URL (e.g. land on the page normally, click a menu item, then click back button) -- we should deactivate all menu items. Otherwise the page scrolls back to the top, hash is empty, but the most recently selected menu item is still active.

Also, this is "Not a Big deal" :wink: but I don't think we need the `click` handler anymore, because the native behavior of clicking a link is to update the hash.

Reply:
Nice!!!!!! :tada: